### ISSUE TITLE
Filling a "Require entry to close a thread" does not immediately enable the option to close it.

### ISSUE THREAD

**Issue by miselico** [osTicket/osTicket#5859]
### Prerequisites

* [ ] Can you reproduce the problem in a fresh installation of the "develop" branch?
     * We are currently running 1.15.1, this issue is not mentioned in the enhancements for the 1.15.2 release
* [ ] Do you have any errors in the PHP error log, or javascript console?
     * No 
* [X] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [X] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description

OSticket now supports "Require entry to close a thread" for fields in the ticket. We have such a filed to internally keep track of additional information. This field is not filled, nor visible to the client.
The field has a number validator.
The issue is that when working on a ticket, and filling a value, the ticket can still not be closed. Only after reloading the ticket page, is it possible to close the ticket.

### Steps to Reproduce

1. Create a form with a field that is
    * enabled
    * not visible for endusers, visible for agents
    * not required for end users, not required for agents
    * not editable by end users, editable by agents
    * data integrity: "Require entry to close a thread" enabled
2. Create a ticket with that form
3. As an operator, fill a value in that filed
4. Try to close the ticket by replying to it.

**Expected behavior:** [What you expected to happen]

I can select the option to close the ticket and the form closes successfully.

**Actual behavior:** [What actually happened]

The option to close the ticket is not in the dropdown menu.
If I now reload the ticket page, then the option is included in the menu.

### Versions

v1.15.1 (fe1d1f8)
Apache/2.4.29 (Ubuntu)

Admin panel -> Dashboard -> Information which also additionally gives you information about your server.

Also, please include the OS and what version of the OS you're running. As well as your browser and browser version.

**Comment by sevmonster** [comment]
I can confirm this is still present on the latest develop head.

**Comment by JediKev** [comment]
@ds6 

Yes, this has been reported a few times. Inline Edit uses AJAX to update and render the value without reloading the page (which is the desired behavior). Since the page isn’t refreshed the statuses at the bottom are not refreshed. You should be able to close it by clicking the change status action as that will fetch an updated list of statuses.

This is on our bug sheet and we will look into this at some point in the future. I have done something similar for "Require Help Topic on Close" so I imagine it’ll be extending that logic for updating a field inline.

Cheers.

**Comment by sevmonster** [comment]
> You should be able to close it by clicking the change status action as that will fetch an updated list of statuses.

Sure, that's what we do. It's only a minor annoyance. I was just clarifying for @miselico's sake since he did not test develop. 🙂
